src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -144,7 +146,9 @@ internal class Scheduler : IScheduler
         /// <summary>
         /// Flag used for debugging by forcing all scheduling to go out-of-proc.
         /// </summary>
-        internal bool ForceAffinityOutOfProc { get; private set; }
+        internal bool ForceAffinityOutOfProc => ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_0)
+            ? InprocNodeDisabledViaEnvironmentVariable || _componentHost.BuildParameters.DisableInProcNode
```

Comment:
Looked at one of the test failures. (9 failed in the group I looked at.) It's setting the env var in the test, so capturing its value at process start makes it effectively unsettable for the test. One option:
```suggestion
            ? (BuildEnvironmentHelper.Instance.RunningTests ? Environment.GetEnvironmentVariable("MSBUILDNOINPROCNODE") == "1" : InprocNodeDisabledViaEnvironmentVariable) || _componentHost.BuildParameters.DisableInProcNode
```

(Not 100% sure if that's available here, but I think it

Reply:
Bah, I ran some UTs but apparently not all I guess

Reply:
If you want to go further, you should be able to eke out a tiny, unimportant perf improvement if you make a function for calling Environment.GetEnvironmentVariable("MSBUILDNOINPROCNODE") == "1" then just cache that. Might even be more readable.

Reply:
`Traits` effecitvely does the `RunningTests` thing, so I'll move the env var there.